coq/loader.ml

```diff
@@ -0,0 +1,65 @@
+(************************************************************************)
+(*         *   The Coq Proof Assistant / The Coq Development Team       *)
+(*  v      *   INRIA, CNRS and contributors - Copyright 1999-2018       *)
+(* <O___,, *       (see CREDITS file for the list of authors)           *)
+(*   \VV/  **************************************************************)
+(*    //   *    This file is distributed under the terms of the         *)
+(*         *     GNU Lesser General Public License Version 2.1          *)
+(*         *     (see LICENSE file for the text of the license)         *)
+(************************************************************************)
+
+(************************************************************************)
+(* Coq serialization API/Plugin                                         *)
+(* Copyright 2016-2019 MINES ParisTech                                  *)
+(* Copyright 2019-2022 Inria                                            *)
+(* Written by Emilio J. Gallego Arias                                   *)
+(************************************************************************)
+[@@@ocamlformat "disable"]
+
+let debug = false
+let list_last l = List.(nth l (length l - 1))
+
+(* Should improve *)
+let map_serlib fl_pkg =
+  let supported = match fl_pkg with
+    (* Supported by serlib *)             (* directory   *)
+    | "coq-core.plugins.ltac"             (* ltac        *)
+    | "coq-core.plugins.firstorder"       (* firstorder  *)
+    | "coq-core.plugins.funind"           (* funind      *)
+    | "coq-core.plugins.ring"             (* ring        *)
+    | "coq-core.plugins.extraction"       (* extraction  *)
+    | "coq-core.plugins.ssrmatching"      (* ssrmatching *)
+    | "coq-core.plugins.ssreflect"        (* ssr *)
+      -> true
+    | _ ->
+      Feedback.msg_warning Pp.(str "Missing serlib plugin: " ++ str fl_pkg);
+      false
+  in
+  if supported
+  then
+    let plugin_name = String.split_on_char '.' fl_pkg |> list_last in
+    Some ("coq-serapi.serlib." ^ plugin_name)
```

Comment:
Does this need to be renamed?

Reply:
No, it does not need renaming. That's a tricky issue as plugins can't register themselves before they are loaded, so for now a manual list needs to be kept.

Reply:
Likely we will add a static init to serlib so at least the list is local.

Reply:
"No it does not need renaming" really means "no, renaming is wrong and would break everything"